api/types/authentication.go

```diff
@@ -535,6 +536,72 @@ func (u *U2F) Check() error {
 			return trace.BadParameter("u2f configuration has an invalid attestation CA: %v", err)
 		}
 	}
+
+	// Removes the port from a URL host.
+	getDomain := func(s string) (string, error) {
+		if strings.Contains(s, ":") {
+			host, _, err := net.SplitHostPort(s)
+			if err != nil {
+				return "", trace.Wrap(err)
+			}
+
+			return host, nil
+		}
+
+		return s, nil
+	}
+
+	// AppID must be a valid URL.
```

Comment:
Can we extract this whole validation to a separate function and document what is considered a valid input? It took me a while to figure out what the "correct input" is.